### PR TITLE
Document Client API Attach and migrate examples

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
## Summary

- rename the user-facing "third-party integration" concept to **Client API Attach mode** for independently managed trainers, while retaining a compatibility page and legacy reference aliases
- document trainer/CJ process ownership, site listener configuration, security, lifecycle, timeouts, troubleshooting, and migration from the legacy agent/exchanger APIs
- clarify that NVFLARE-managed integrations such as Flower use their dedicated Controller/Executor and TIE path rather than Attach
- migrate current architecture pages, tutorials, Docker jobs, CC provisioning, and experiment-tracking examples to the unified `ClientAPIExecutor` modes

## Why

"Third-party integration" conflated library/vendor integration with process ownership. Attach is specifically for a trainer process started and owned outside NVFLARE that initiates a connection to the Client Job. The documentation and examples need to establish that contract before the legacy components are removed separately.

## User impact

Users now choose `in_process`, `external_process`, or `attach` based on trainer-process ownership. The runnable Attach example demonstrates an independently started trainer and provides shared-file, mTLS, security, and component-policy guidance.

## Validation

- `./runtest.sh -s --skip-install` in an isolated environment
- full Sphinx HTML build with no warnings from changed pages
- shared-filesystem Attach POC end to end: independently started trainer, three completed rounds, `FINISHED:COMPLETED`, expected final model
- Black, isort, Flake8, Python compilation, and `git diff --check`
- migrated JSON, YAML, and notebook parsing
- construction of all migrated `ClientAPIExecutor` configurations

## Not run

- production mTLS deployment
- full GPU Docker/DDP execution; the migrated executor configurations were constructed and validated

### CODE REVIEW

File: web/src/components/overview.astro [hunk 1]
```diff
@@ -50,14 +50,14 @@ const features = [
...
+    title: "Client API Attach Mode",
     icon: `<svg class="stroke-nvidia mt-2 ml-4 w-10 h-10 text-gray-800" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="24" height="24" fill="none" viewBox="0 0 24 24">
       <path stroke="" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M15 17h3a3 3 0 0 0 0-6h-.025a5.56 5.56 0 0 0 .025-.5A5.5 5.5 0 0 0 7.207 9.021C7.137 9.017 7.071 9 7 9a4 4 0 1 0 0 8h2.167M12 19v-9m0 0-2 2m2-2 2 2"/>
     </svg>
     `,
     description:
-      "Enables a seamless integration with external 3rd-party systems with the FlareAgent to receive tasks and submit results to the NVIDIA FLARE server.",
-    link: `https://nvflare.readthedocs.io/en/${gh_branch}/programming_guide/execution_api_type/3rd_party_integration.html#3rd-party-integration`,
+      "Lets independently started and externally owned trainers attach to an NVFlare Client Job to receive tasks and submit results without transferring process ownership.",
+    link: `https://nvflare.readthedocs.io/en/${gh_branch}/programming_guide/execution_api_type/client_api_attach.html#client-api-attach`,
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply web/src/components/overview.astro:60]: Verified with a local Sphinx HTML build: the explicit RST label renders as `id="client-api-attach"`, so the existing `#client-api-attach` fragment is correct and no code change is needed.

File: docs/user_guide/admin_guide/configurations/logging_configuration.rst [hunk 2]
```diff
@@ -186,7 +186,7 @@ Example configuration and output:
 
 .. code-block:: json
 
-    {"asctime": "2025-01-14 14:44:46,559", "name": "PTInProcessClientAPIExecutor", "fullName": "nvflare.app_opt.pt.in_process_client_api_executor.PTInProcessClientAPIExecutor", "levelname": "INFO", "fl_ctx": "[identity=site-1, run=fc711945-a7cf-4834-9fc4-aa9cb60e327b, peer=example_project, peer_run=fc711945-a7cf-4834-9fc4-aa9cb60e327b, task_name=train, task_id=a16b7a02-b2ea-4eb5-895a-b40d507b2c5c]", "message": "execute for task (train)"}
+    {"asctime": "2025-01-14 14:44:46,559", "name": "ClientAPIExecutor", "fullName": "nvflare.app_common.executors.client_api_executor.ClientAPIExecutor", "levelname": "INFO", "fl_ctx": "[identity=site-1, run=fc711945-a7cf-4834-9fc4-aa9cb60e327b, peer=example_project, peer_run=fc711945-a7cf-4834-9fc4-aa9cb60e327b, task_name=train, task_id=a16b7a02-b2ea-4db8-9fc4-aa9cb60e327b]", "message": "execute for task (train)"}
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply docs/user_guide/admin_guide/configurations/logging_configuration.rst]: Fixed in `ebcc65ddc`: the JSON formatter example now uses the same `task_id` as the preceding base formatter output.

File: docs/programming_guide/timeouts.rst [hunk 3]
```diff
@@ -2979,7 +2761,7 @@ job to fail.
 2. **Heartbeat Relationships**: Always maintain proper ratios
    
    - ``heartbeat_timeout`` = 3-6x ``heartbeat_interval``
-   - ``agent_heartbeat_timeout`` = 3-6x ``agent_heartbeat_interval``
+   - ``heartbeat_timeout`` = 3-6x ``heartbeat_interval`` for out-of-process Client API modes
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply docs/programming_guide/timeouts.rst]: Fixed in `ebcc65ddc`: the duplicate guidance is now one heartbeat-ratio bullet that explicitly includes out-of-process Client API modes.

File: docs/user_guide/admin_guide/configurations/variable_resolution.rst [hunk 4]
```diff
@@ -206,7 +207,8 @@ Before discussing this advanced topic, let's first show an example of job config
    }
 
 
-This job requires two pipes, one for task exchange (task_pipe), another for metrics collection (metric_pipe).
+This illustrative job declares two application pipes, ``task_pipe`` and
+``metric_pipe``, in addition to its Client API executor.
 If you look at their configuration closely, you will see that: there are many args to configure, and the configs of the two pipes are identical except for their ``id`` values. It is tedious and error-prone to configure many args in multiple places.
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply docs/user_guide/admin_guide/configurations/variable_resolution.rst]: Fixed in `4230b453a`: the legacy CellPipe example was replaced entirely with a neutral parameterized `ConvertToFedEvent` component example, so it no longer implies that Client API requires user-defined pipes.

File: examples/advanced/cc_provision/jobs/hello-pt_cifar10_fedavg/app_site-1/config/config_fed_client.json [hunk 5]
```diff
@@ -6,9 +6,13 @@
                 "*"
             ],
             "executor": {
-                "path": "nvflare.app_opt.pt.in_process_client_api_executor.PTInProcessClientAPIExecutor",
+                "path": "nvflare.app_common.executors.client_api_executor.ClientAPIExecutor",
```
Comment (nvkevlu) [inline examples/advanced/cc_provision/jobs/hello-pt_cifar10_fedavg/app_site-1/config/config_fed_client.json]: [P1] The CC image executing this config still installs nvflare~=2.7.1rc in examples/advanced/cc_provision/docker/Dockerfile:9. That constraint remains on 2.7.x, where ClientAPIExecutor does not exist. The Client Job will therefore fail while constructing this executor, before /local/custom/client.py starts. Please install the current NVFlare source/a release containing ClientAPIExecutor, or retain PTInProcessClientAPIExecutor for this pinned image.

File: docs/programming_guide/timeouts.rst [hunk 6]
```diff
@@ -460,235 +440,59 @@ Session management for programmatic API (flare_api.py):
...
-     - Time to wait for external process before ``flare.init()`` call
-
-ClientAPILauncherExecutor
--------------------------
+Client API Executor Timeouts
+============================
 
-The Client API executor extends base timeouts with more conservative defaults
-(client_api_launcher_executor.py:29-53):
+``ClientAPIExecutor`` exposes one timeout surface across its three execution
```
Comment (nvkevlu) [inline docs/programming_guide/timeouts.rst:446]: [P2] This migration leaves later timeout guidance inconsistent with the new execution path. For example, lines 1362–1376 and 1402–1409 still recommend submit_result_timeout, download_complete_timeout, PEER_READ_TIMEOUT, and max_resends for ScriptRunner(launch_external_process=True). That now selects ClientAPIExecutor and its Cell-based trainer API, which does not consume these legacy Pipe/FlareAgent settings, so the configuration is silently ineffective. Could these examples use supported ClientAPIExecutor timeouts—exposing them through the recipe if necessary—or be explicitly scoped to BaseScriptRunner/ClientAPILauncherExecutor? Similar guidance remains in tensor_downloader.rst and timeout_troubleshooting.rst.

File: docs/examples/hello_tf_job_api.rst [hunk 7]
```diff
@@ -175,9 +175,13 @@ The client configuration is ``config_fed_client.json`` in the config folder of e
                "*"
             ],
             "executor": {
-               "path": "nvflare.app_opt.tf.in_process_client_api_executor.TFInProcessClientAPIExecutor",
+               "path": "nvflare.app_common.executors.client_api_executor.ClientAPIExecutor",
                "args": {
-                  "task_script_path": "src/hello-tf_fl.py"
+                  "execution_mode": "in_process",
+                  "task_script_path": "src/hello-tf_fl.py",
```
Comment (nvkevlu) [inline docs/examples/hello_tf_job_api.rst]: [P2] This still is not the configuration exported by the linked examples/hello-world/hello-tf example. Its job.py uses train_script="client.py" and model=Net(), while src/hello-tf_fl.py, src.tf_net.TFNet, and fedavg_script_runner_hello-tf.py do not exist. Could we regenerate/update the entire page rather than only replacing the executor class? The expected paths are client.py and model.Net.

File: docs/programming_guide/timeouts.rst [hunk 8]
```diff
@@ -460,235 +440,59 @@ Session management for programmatic API (flare_api.py):
...
-   * - log_pull_interval
-     - None
-     - How often to pull logs (None = same as result_pull_interval)
+     - Interval between session heartbeats.
+   * - ``heartbeat_timeout``
+     - ``external_process``, ``attach``
+     - 30.0
+     - Session lease on missed heartbeats; active payload transfer preserves
+       liveness.
+   * - ``task_wait_timeout``
```
Comment (nvkevlu) [inline docs/programming_guide/timeouts.rst:478]: [P2] task_wait_timeout has no effect in in_process mode—the in-process backend never reads this field. Please scope this row to external_process and attach so users do not rely on an ignored timeout.

File: examples/advanced/cc_provision/jobs/hello-pt_cifar10_fedavg/app_site-1/config/config_fed_client.json (hunk 5 above)
Comment (YuanTingHsieh) [reply examples/advanced/cc_provision/jobs/hello-pt_cifar10_fedavg/app_site-1/config/config_fed_client.json]: Fixed in `48e42afc7`: the CC image keeps its existing 2.7.x pin, and the job configuration now retains the compatible `PTInProcessClientAPIExecutor`. I also removed the now-unneeded `ClientAPIExecutor` class allow-list entry.

File: docs/programming_guide/timeouts.rst (hunk 6 above)
Comment (YuanTingHsieh) [reply docs/programming_guide/timeouts.rst:446]: Fixed in `48e42afc7`: the timeout, tensor-downloader, and troubleshooting docs now distinguish the Cell-based `ClientAPIExecutor` path from the legacy `BaseScriptRunner` / `ClientAPILauncherExecutor` Pipe path. The ineffective `submit_result_timeout`, `download_complete_timeout`, `PEER_READ_TIMEOUT`, and `max_resends` recommendations were removed from the new executor examples and explicitly scoped to the legacy stack.

File: docs/examples/hello_tf_job_api.rst (hunk 7 above)
Comment (YuanTingHsieh) [reply docs/examples/hello_tf_job_api.rst]: Fixed in `48e42afc7`: the page now matches the current `examples/hello-world/hello-tf` sources and generated configuration, including `job.py`, `client.py`, `model.Net`, and `task_script_path="client.py"`; the obsolete script, model, and `src/` references were removed.

File: docs/programming_guide/timeouts.rst (hunk 8 above)
Comment (YuanTingHsieh) [reply docs/programming_guide/timeouts.rst:478]: Fixed in `48e42afc7`: the `task_wait_timeout` row is now scoped to `external_process` and `attach`, matching the backends that consume it.

File: docs/examples/hello_tf_job_api.rst [hunk 9]
```diff
@@ -3,229 +3,140 @@
...
-      "task_result_filters": []
+     "tasks": ["*"],
+     "executor": {
+       "path": "nvflare.app_common.executors.client_api_executor.ClientAPIExecutor",
+       "args": {
+         "execution_mode": "in_process",
+         "task_script_path": "client.py",
+         "params_exchange_format": "keras_layer_weights",
+         "server_expected_format": "numpy",
+         "train_with_evaluation": true
```
Comment (nvkevlu) [inline docs/examples/hello_tf_job_api.rst]: [P2] The current recipe does not generate train_with_evaluation: true. ScriptRunner constructs ClientAPIExecutor without this argument, so its effective value is the generic default, false, and the exported JSON omits it. The changed Hello PyTorch snippet has the same issue. Please either preserve the previous framework-specific true behavior or correct the snippets.

File: docs/examples/hello_tf_job_api.rst [hunk 10]
```diff
@@ -3,229 +3,140 @@
...
-
-Let's get started. Since this task is using TensorFlow, let's go ahead and install the library inside our virtual environment:
+From the repository root, create an environment, install the example
+requirements, and run the job recipe:
 
 .. code-block:: shell
 
-  (nvflare-env) $ python3 -m pip install tensorflow
+   $ cd examples/hello-world/hello-tf
+   $ python3 -m pip install -r requirements.txt
```
Comment (nvkevlu) [inline docs/examples/hello_tf_job_api.rst:24]: [P2] This command installs nvflare~=2.7.2rc from the example requirements, which restricts NVFlare to 2.7.x. That version does not contain ClientAPIExecutor, so following these instructions cannot produce the unified executor configuration documented below. The changed Hello PyTorch and Swarm pages have the same version mismatch. Please update the requirements/install instructions or retain the 2.7-compatible configuration.

File: docs/programming_guide/timeouts.rst [hunk 11]
```diff
@@ -1553,21 +1359,21 @@ Framework-level settings for large payload transfers (fl_constant.py:553, comm_c
...
-- ``PEER_READ_TIMEOUT`` should be at least the configured streaming per-request
-  timeout so the parent client job does not resend the task while the subprocess
-  is still downloading a large payload.
-- ``download_complete_timeout`` should be at least the configured streaming
-  per-request timeout and long enough for the server to pull large tensor
-  results from the subprocess after result ACK.
-- ``max_resends`` should stay finite. The recipe default is ``3``; raise it
-  only when the network is expected to recover after a small number of delayed
-  result acknowledgments.
+- ``task_wait_timeout`` and ``result_wait_timeout`` bound task acceptance and
```
Comment (nvkevlu) [inline docs/programming_guide/timeouts.rst]: [P2] Please scope this progress-aware claim. ExternalProcessBackend extends TASK_READY waiting while a transfer is active, but AttachBackend uses an absolute task_wait_timeout deadline without that progress callback. A slow but active Attach task download can therefore still exhaust task_wait_timeout. The same broad statement appears in the changed Tensor Downloader paragraph.

File: docs/examples/hello_tf_job_api.rst [hunk 12]
```diff
@@ -3,229 +3,140 @@
...
-
-.. code-block:: python
+.. code-block:: shell
 
-   # Define the initial global model and send to server
-   job.to(TFNet(), "server")
+   $ python3 job.py --n_clients 3 --num_rounds 5
 
+The recipe executes with :class:`SimEnv<nvflare.recipe.sim_env.SimEnv>`. Results
+and logs are written under ``/tmp/nvflare/jobs/workdir`` by default.
```
Comment (nvkevlu) [inline docs/examples/hello_tf_job_api.rst]: SimEnv writes under /tmp/nvflare/simulation, not /tmp/nvflare/jobs/workdir

File: examples/docker/README.md [hunk 13]
```diff
@@ -29,6 +29,16 @@ nvflare provision -p project.yml
 
 This generates a workspace under `workspace/docker_test_project/` relative to the current directory.
 
+Before preparing the client kits, copy each client's
```
Comment (nvkevlu) [inline examples/docker/README.md]: the instructions edit resources.json, but deploy prepare reads resources.json.default and removes the active resources.json from the prepared kit, so that edit has no effect.

Comment (nvkevlu) [reply examples/docker/README.md]: Execution testing confirms that the documented edit does not reach the prepared kit. More importantly, these jobs have byoc: true, and ComponentPathAuthorizer bypasses class authorization for BYOC jobs. The literal hello-numpy workflow therefore completed even though the prepared default did not contain ClientAPIExecutor. This step should likely be removed rather than changed to edit resources.json.default.

**Comment by nvkevlu** [comment]
Tested commit 48e42afc7 on a four-RTX-A4000 Colossus host. The literal NumPy job completed. The unmodified PyTorch job failed because UID:GID 36567:30 could not create /var/tmp/nvflare/data; after a data-directory-only workaround, both in-process PyTorch and external-process DDP completed. The DDP run validated ClientAPIExecutor plus torchrun, but both logical sites were assigned the same GPU UUID pair, so it should not be described as a four-distinct-GPU test.

File: docs/examples/hello_tf_job_api.rst (hunk 10 above)
Comment (YuanTingHsieh) [reply docs/examples/hello_tf_job_api.rst:24]: we can change all example requirement later for all after 2.9.x rc is ready

File: docs/examples/hello_tf_job_api.rst (hunk 9 above)
Comment (YuanTingHsieh) [reply docs/examples/hello_tf_job_api.rst]: Fixed in `1554dce07`: removed `train_with_evaluation: true` from both the TensorFlow and PyTorch generated-config snippets so they match the current `ScriptRunner` export and the `ClientAPIExecutor` default.

File: docs/programming_guide/timeouts.rst (hunk 11 above)
Comment (YuanTingHsieh) [reply docs/programming_guide/timeouts.rst]: Fixed in `1554dce07`: both timeout pages now state that active task-transfer progress extends `task_wait_timeout` only for `external_process`; Attach is documented as using an absolute deadline covering task download and trainer acceptance. Result publication and subsequent payload-streaming semantics are also separated.

File: examples/docker/README.md (hunk 13 above)
Comment (YuanTingHsieh) [reply examples/docker/README.md]: Fixed in `1554dce07`: removed the class-allow-list preparation step entirely. These Docker example jobs use `byoc: true`, so `ComponentPathAuthorizer` does not require adding `ClientAPIExecutor`, and editing the pre-prepare `resources.json` was ineffective.

File: docs/examples/hello_tf_job_api.rst (hunk 12 above)
Comment (YuanTingHsieh) [reply docs/examples/hello_tf_job_api.rst]: Fixed in `1554dce07`: the documented `SimEnv` results/log location is now `/tmp/nvflare/simulation`.

**Review by nvkevlu — APPROVED** [review]
If we are planning to update requirements in a different PR then this LGTM
